### PR TITLE
fix(ci): bump macos runners to 14 and explcitly specify targets on macos

### LINKED ISSUE #269: Unable to import fastexcel - symbol not found in flat namespace '_PyPyBaseObject_Type'
## Description

**Platform:** M3 MacBook Pro running macOS 14.5
**Python version:** 3.12.4
**fastexcel version:** 0.11.3

**Issue:** I am completely unable to import fastexcel. When using a clean virtual environment, I install fastexcel using pip, then attempt to import it. When I do so, I get the error `ImportError: symbol not found in flat namespace '_PyPyBaseObject_Type'`. I believe this is an issue new to 0.11.3.

### 1. Create a new virtual environment and install fastexcel
```shell
python3 -m venv venv
source venv/bin/activate
pip install fastexcel
```

### 2. Attempt to import fastexcel
```python
import fastexcel
```

This is the error message I'm seeing:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/cmhack0114/Github/christopher-hacker/fastexcel-test/venv/lib/python3.12/site-packages/fastexcel/__init__.py", line 20, in <module>
    from ._fastexcel import (
ImportError: dlopen(/Users/cmhack0114/Github/christopher-hacker/fastexcel-test/venv/lib/python3.12/site-packages/fastexcel/_fastexcel.abi3.so, 0x0002): symbol not found in flat namespace '_PyPyBaseObject_Type'
```

### Temporary solution

Downgrading to 0.11.2 fixed the problem for me.

### PR DESCRIPTION
**Pull request by lukapeschke**
fix #269

### CODE REVIEW

**Review by PrettyWood — APPROVED** [review]

**Comment by lukapeschke** [comment]
waiting for user feedback on #269 before I merge this

**Pull request merged**
